functions/classes/class.Subnets.php

```diff
@@ -828,6 +828,79 @@ public function get_ipv4_masks () {
 	*/
 
 	/**
+	 * Deletes temporary table containing slave ids for a given subnetId
+	 *
+	 * @access public
+	 * @param int $subnetId
+	 * @return void
+	 */
+	public function reset_subnet_familytree(int $subnetId) {
+	  try {
+	    $this->Database->runQuery("DROP TABLE IF EXISTS tmp_subnet_familytree_${subnetId};");
+	  }
+	  catch (Exception $e) {
+	    $this->Result->show("danger", _("Error: ").$e->getMessage());
+	  }
+	}
+
+	/**
+	 * Generates temporary table containing slave ids for a given subnetId
+	 *
+	 * @access public
+	 * @param int $subnetId
+	 * @return void
+	 */
+	public function create_subnet_familytree(int $subnetId) {
+
+	  // MySQL does not support multiple references to a temporary table in the same query.
+	  // Ideally we would populate temp_table with our initial subnetId and then run the query below until
+	  // the inserted record count is zero (no more slaves ids found).
+	  //
+	  //  INSERT INTO temp_table SELECT id FROM subnets WHERE id NOT IN (temp_table) AND masterSubnetId IN (temp_table)
+	  //
+	  // Work around the 'wont-fix' limitation by using a temporary table per iteration and consolidate the results.
+
+	  try {
+	    $this->reset_subnet_familytree($subnetId);
+
+	    $rowCount = 0; $level = 0;
+
+	    $this->Database->runQuery("DROP TABLE IF EXISTS tmp_subnet_familytree_${subnetId}_${level};");
+
+	    $query = "CREATE TEMPORARY TABLE tmp_subnet_familytree_${subnetId}_${level} (id int(11) PRIMARY KEY) ENGINE = MEMORY
```

Comment:
What happens if user has no support for ENGINE MEMORY type ? Maybe we can add new option in config.php to set memory type and have MEMORY as default?